x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/detection_rules_client/methods/patch_rule.ts

```diff
@@ -39,16 +48,28 @@ export const patchRule = async (
 
   const patchedRule = convertPatchAPIToInternalSchema(nextParams, existingRule);
 
-  const update = await rulesClient.update({
+  const patchedInternalRule = await rulesClient.update({
     id: existingRule.id,
     data: patchedRule,
   });
 
-  await toggleRuleEnabledOnUpdate(rulesClient, existingRule, nextParams.enabled);
+  const { enabled } = await toggleRuleEnabledOnUpdate(
+    rulesClient,
+    existingRule,
+    nextParams.enabled
+  );
```

Comment:
Can this function throw exceptions? If yes, I'd suggest to think about if we should add error handling to it so it's safe to call.

The case I'm thinking about is: if the `await rulesClient.update` call above was successful, how should we handle a failure of the subsequent enable/disable call?